flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/TwoPhaseCommitSinkFunction.java

```diff
@@ -361,5 +376,247 @@ public void setPendingCommitTransactions(List<TXN> pendingCommitTransactions) {
 		public void setContext(Optional<CONTEXT> context) {
 			this.context = context;
 		}
+
+		@Override
+		public boolean equals(Object o) {
+			if (this == o) {
+				return true;
+			}
+			if (o == null || getClass() != o.getClass()) {
+				return false;
+			}
+
+			State<?, ?> state = (State<?, ?>) o;
+
```

Comment:
Not a deal breaker but these if's are a bit too complicated.
Using `Objects.equals()` on each of the components and `&&` would simplify them.

Reply:
I used IntelliJ to generate those. :sweat_smile: